booster/tools/booster/Proxy.hs

```diff
@@ -384,43 +424,38 @@ respondEither cfg@ProxyConfig{statsVar, boosterState} booster kore req = case re
-                                                        , fallbackLog
-                                                        ]
-                                            executionLoop
-                                                logSettings
-                                                def
-                                                ( currentDepth + boosterResult.depth + koreResult.depth
-                                                , time + bTime + kTime
-                                                , koreTime + kTime
-                                                , accumulatedLogs
-                                                )
-                                                r{ExecuteRequest.state = execStateToKoreJson koreResult.state}
+                                            continueWith [] (execStateToKoreJson koreResult.state)
                                         _ -> do
                                             -- otherwise we have hit a different
                                             -- HaltReason, at which point we should
                                             -- return, setting the correct depth
                                             Log.logInfoNS "proxy" . Text.pack $
                                                 "Kore " <> show koreResult.reason <> " at " <> show koreResult.depth
-                                            logStats ExecuteM (time + bTime + kTime, koreTime + kTime)
-                                            pure $
-                                                Right $
-                                                    Execute
-                                                        koreResult
-                                                            { depth = currentDepth + boosterResult.depth + koreResult.depth
-                                                            , logs =
-                                                                combineLogs
-                                                                    [ rpcLogs
-                                                                    , boosterResult.logs
-                                                                    , koreResult.logs
-                                                                    , fallbackLog
-                                                                    ]
-                                                            }
+                                            -- perform post-exec simplification.
+                                            -- NB This has been found to make a difference,
+                                            -- even for results produced by legacy-kore.
+                                            postExecResult <-
+                                                simplifyExecResult logSettings r._module def koreResult
```

Comment:
We *may* want to get away without the simplification if we made more a step in Kore (i.e. `Brandhing` at depth 1, rather than 0). We should consider trying that in case we see a degradation in performance due to the additional simplify requests.